graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH5.kt

```diff
@@ -0,0 +1,126 @@
+package com.anaplan.engineering.azuki.graphs.eacs
+
+import com.anaplan.engineering.azuki.core.runner.Eac
+import com.anaplan.engineering.azuki.core.system.BEH
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphBehaviours
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphFunctionalElements
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+@BEH(GraphBehaviours.GetShortestPath, GraphFunctionalElements.GraphFunction, """
+
+""")
+class BEH5 : GraphScenario() {
+
+    @Eac("""
+        Given a graph that contains at least path from a beginning vertex to an ending vertex,
+        the graph will have a shortest path between these vertices
+    """)
+    fun onePath() {
+        given {
+            thereIsAnUndirectedGraph(graphA) {
+                edge("a", "b")
+            }
+        }
+        then {
+            hasShortestPath(graphA, "a", "b")
+        }
+    }
+
+    @Eac("""
+        Given a graph that contains more then one path between two vertices,
+        there is a shortest path
+    """, """
+        If two paths have the same number of vertices and are both the shortest only
+        the first path defined will be recognised as the shortest path.
+    """)
+    fun multiplePaths() {
+        given {
+            thereIsAnUndirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("c", "d")
+                edge("d", "e")
+                edge("a", "f")
+                edge("f", "e")
+            }
+        }
+        then {
+            hasShortestPath(graphA, "a", "f", "e")
+        }
+    }
+
+    @Eac("""
+        Given a directed graph,
+        the shortest path wont include an edge that is declared in the opposite direction to the path
```

Comment:
🚫 Not an AC for this function. If an edge goes against the direction of the other edges in a path, then it cannot be added to that path by definition.

This is nonetheless a useful analysis scenario.